lib/node_modules/@stdlib/number/float64/base/exponent/src/exponent.c

```diff
@@ -0,0 +1,41 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/number/float64/base/exponent.h"
+#include "stdlib/number/float64/base/get_high_word.h"
+#include <stdint.h>
+
+/**
+* Extracts the exponent from a double-precision floating-point number.
+*
+* @param x       input value
+* @param out     output exponent
+*
+* @example
+* #include <stdint.h>
+*
+* int64_t out;
+* stdlib_base_float64_exponent( 4.0, &out );
+*/
+void stdlib_base_float64_exponent( const double x, int64_t *out ) {
```

Comment:
A 32-bit int should be fine here. For float64, the exponent fits within a 16-bit int, but using `int32` probably makes more sense in terms of common usage.